core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -65,10 +66,61 @@ int HIP::impl_is_initialized() {
 }
 
 void HIP::impl_initialize(InitializationSettings const& settings) {
+  const int hip_device_id = Impl::get_gpu(settings);
+
+  // Need at least a GPU device
+  int hipDevCount;
+  KOKKOS_IMPL_HIP_SAFE_CALL(hipGetDeviceCount(&hipDevCount));
```

Comment:
Seeing that `HIP::detect_device_count()` now does the same thing as this line,

```suggestion
  int hipDevCount = HIP::detect_device_count();
```